From 73f9671edecbdee831761aea079edcef7c086f79 Mon Sep 17 00:00:00 2001
From: Alberto Ponces <ponces26@gmail.com>
Date: Fri, 1 Mar 2024 10:49:00 +0000
Subject: [PATCH] Add "android.permission.MONITOR_INPUT" to manifest

Attempt to resolve:
11-18 07:16:36.964  9447  9447 E AndroidRuntime: FATAL EXCEPTION: main
11-18 07:16:36.964  9447  9447 E AndroidRuntime: Process: com.android.wallpaper, PID: 9447
11-18 07:16:36.964  9447  9447 E AndroidRuntime: java.lang.SecurityException: Permission Denial: opening provider com.google.android.apps.wallpaper.provider.RecentWallpapersProvider from ProcessRecord{152cdeb 9447:com.android.wallpaper/u0a138} (pid=9447, uid=10138) requires android.permission.MONITOR_INPUT or android.permission.MONITOR_INPUT
11-18 07:16:36.964  9447  9447 E AndroidRuntime: 	at android.os.Parcel.createExceptionOrNull(Parcel.java:3057)
11-18 07:16:36.964  9447  9447 E AndroidRuntime: 	at android.os.Parcel.createException(Parcel.java:3041)
11-18 07:16:36.964  9447  9447 E AndroidRuntime: 	at android.os.Parcel.readException(Parcel.java:3024)
11-18 07:16:36.964  9447  9447 E AndroidRuntime: 	at android.os.Parcel.readException(Parcel.java:2966)
11-18 07:16:36.964  9447  9447 E AndroidRuntime: 	at android.content.IContentService$Stub$Proxy.registerContentObserver(IContentService.java:1032)
11-18 07:16:36.964  9447  9447 E AndroidRuntime: 	at android.content.ContentResolver.registerContentObserver(ContentResolver.java:2728)
11-18 07:16:36.964  9447  9447 E AndroidRuntime: 	at android.content.ContentResolver.registerContentObserver(ContentResolver.java:2676)
11-18 07:16:36.964  9447  9447 E AndroidRuntime: 	at com.android.wallpaper.picker.customization.data.content.WallpaperClientImpl$recentWallpapers$1.invokeSuspend(WallpaperClientImpl.kt:74)
11-18 07:16:36.964  9447  9447 E AndroidRuntime: 	at com.android.wallpaper.picker.customization.data.content.WallpaperClientImpl$recentWallpapers$1.invoke(Unknown Source:8)
11-18 07:16:36.964  9447  9447 E AndroidRuntime: 	at com.android.wallpaper.picker.customization.data.content.WallpaperClientImpl$recentWallpapers$1.invoke(Unknown Source:4)
11-18 07:16:36.964  9447  9447 E AndroidRuntime: 	at kotlinx.coroutines.flow.ChannelFlowBuilder.collectTo$suspendImpl(Builders.kt:322)
11-18 07:16:36.964  9447  9447 E AndroidRuntime: 	at kotlinx.coroutines.flow.ChannelFlowBuilder.collectTo(Unknown Source:0)
11-18 07:16:36.964  9447  9447 E AndroidRuntime: 	at kotlinx.coroutines.flow.CallbackFlowBuilder.collectTo(Builders.kt:336)
11-18 07:16:36.964  9447  9447 E AndroidRuntime: 	at kotlinx.coroutines.flow.internal.ChannelFlow$collectToFun$1.invokeSuspend(ChannelFlow.kt:60)
11-18 07:16:36.964  9447  9447 E AndroidRuntime: 	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
11-18 07:16:36.964  9447  9447 E AndroidRuntime: 	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
11-18 07:16:36.964  9447  9447 E AndroidRuntime: 	at kotlinx.coroutines.internal.LimitedDispatcher.run(LimitedDispatcher.kt:42)
11-18 07:16:36.964  9447  9447 E AndroidRuntime: 	at kotlinx.coroutines.scheduling.TaskImpl.run(Tasks.kt:95)
11-18 07:16:36.964  9447  9447 E AndroidRuntime: 	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:570)
11-18 07:16:36.964  9447  9447 E AndroidRuntime: 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:750)
11-18 07:16:36.964  9447  9447 E AndroidRuntime: 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:677)
11-18 07:16:36.964  9447  9447 E AndroidRuntime: 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:664)
11-18 07:16:36.964  9447  9447 E AndroidRuntime: 	Suppressed: kotlinx.coroutines.DiagnosticCoroutineContextException: [StandaloneCoroutine{Cancelling}@1c974aa, Dispatchers.Main.immediate]

---
 AndroidManifest.xml | 1 +
 1 file changed, 1 insertion(+)

diff --git a/AndroidManifest.xml b/AndroidManifest.xml
index 6faa016f..f91fa809 100755
--- a/AndroidManifest.xml
+++ b/AndroidManifest.xml
@@ -10,6 +10,7 @@
     <uses-permission android:name="android.permission.SET_WALLPAPER_COMPONENT" />
     <uses-permission android:name="android.permission.READ_DEVICE_CONFIG" />
     <uses-permission android:name="android.permission.MODIFY_DAY_NIGHT_MODE" />
+    <uses-permission android:name="android.permission.MONITOR_INPUT" />
     <uses-permission android:name="android.permission.CUSTOMIZE_SYSTEM_UI" />
     <uses-permission android:name="android.permission.BIND_WALLPAPER"/>
     <uses-permission android:name="android.permission.READ_MEDIA_IMAGES"/>
-- 
2.34.1

